### PR TITLE
Feature/pet application data sa

### PR DESCRIPTION
**Pull request by StevieAmb**
### 1. What changed?
- The application data for individual pet ids are queried, and then passed down through the Applications Container, and then iterated through in order to create application cards based on the queried data for each pet. 

### 2. Why is this change necessary?
- This change is necessary because the user(re-homer person), is going to need to see a list of how many users (adopter person), have filled out an application for the pet they put up to re-home.

### 3. What changed technically that may impact others?
- Well, The data for the applications have been queried, and so the backend now can be visually shown the images that will still be needed to complete the seeded data file in a way needed by the front-end to successfully populate the home page.

### 4. How do we test it?
- When clicking an individual pet from the home page, you can click the "View Applications" button, and see the list of applications filled out for that pet.

### CODE REVIEW

File: cypress/fixtures/petApplicationsPage.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (anewb87) [inline cypress/fixtures/petApplicationsPage.json:1]: Nice looking fixture. Gotta love some stubbed data.

File: cypress/integration/pet_applications_spec.js [hunk 2]
```diff
@@ -1,20 +1,37 @@
...
+    cy.intercept(
+      {
+        method: 'POST',
+        url: 'https://not-fur-gotten-be.herokuapp.com/graphql',
+        headers: {
+          'x-gql-operation-name': 'getPetById',
+        }
+      },
+      {
+        fixture: 'petApplicationsPage.json'
```
Comment (anewb87) [inline cypress/integration/pet_applications_spec.js:12]: Intercepted and stubbed.

File: src/Components/App.js [hunk 3]
```diff
@@ -9,14 +9,15 @@ import ApplicationFormModal from './ApplicationFormModal';
 import DonationFormModal from './DonationFormModal';
 
 function App() {
+  
   return (
     <div className="App">
       <Routes>
         <Route path="/"  element={<Home />} />
         <Route path="pet/:id" element={<Pet />} />
+        <Route path="pet/:id/applications" element={<PetApplications />} />
```
Comment (anewb87) [inline src/Components/App.js:18]: Simply moved it up for developer empathy.

File: src/Components/ApplicationCard.js [hunk 4]
```diff
@@ -1,16 +1,16 @@
 import React from 'react';
 import '../Styles/ApplicationCard.scss';
 
-const ApplicationCard = () => {
+const ApplicationCard = ({name, email, description}) => {
```
Comment (anewb87) [inline src/Components/ApplicationCard.js:4]: Pulling in the real data! This is exciting. And it works. Which is even more exciting.

File: src/Components/ApplicationsContainer.js [hunk 5]
```diff
@@ -2,13 +2,21 @@ import React from 'react';
...
+  let allApplications = apps.apps.applications.map(application => {
+    return (
+      <ApplicationCard
+          name={application.name}
+          email={application.email}
+          description={application.description}
+      />
+    )
+  })
+  let display = allApplications.length ? allApplications : <p className='no-applications-section'>No applications quite yet.</p>
```
Comment (anewb87) [inline src/Components/ApplicationsContainer.js:16]: Good catch on the conditional. If there aren't any applications we don't want the user to be left with a blank page.

File: src/Components/PetApplications.js [hunk 6]
```diff
@@ -1,14 +1,42 @@
...
+import { gql, useQuery } from '@apollo/client';
 import '../Styles/PetApplications.scss';
 
+
+const GET_PET_APPLICATION = gql`
+query getPetById($id: ID!) {
+  getPetById(id: $id) 
+  {
+    name
+    applications {
```
Comment (anewb87) [inline src/Components/PetApplications.js:14]: Query to get applications for specific pet.

File: cypress/fixtures/petApplicationsPage.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+              },
+              {
+                  "name": "April",
+                  "email": "april@yahoo.com",
+                  "description": "I used to have a cat that looks like Garfield.  I am ready for a cat again and would love to adopt.  I have a large home where I can set up a big cat tree for Garfield so he can watch birds out the window."
+              }
+          ]
+      }
+  }
+}
```
Comment (Fordo29) [inline cypress/fixtures/petApplicationsPage.json:19]: great work!  I liked how you changed the values to ensure the tests are intercepted appropriately.

File: src/Components/PetApplications.js [hunk 8]
```diff
@@ -1,14 +1,42 @@
...
 const PetApplications = () => {
+  let { id } = useParams();
+  const { loading, error, data } = useQuery(GET_PET_APPLICATION, {variables: {id}})
+
+
+  if (loading) {
+    return <p>Loading...</p>
+  }
+  if (error) {
+    return <p>{error}</p>
```
Comment (anewb87) [inline src/Components/PetApplications.js:32]: We've set it up the error like this for all our queries. I'm thinking we'll want to redirect to our Error Page at some point.

**Review by anewb87 — APPROVED** [review]
Nice! Making queries and putting real data in.

**Review by ectweitmann — APPROVED** [review]

**Review by Fordo29 — APPROVED** [review]

File: cypress/fixtures/zeroPetApplications.json [hunk 9]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (anewb87) [inline cypress/fixtures/zeroPetApplications.json:1]: Nice! Two different fixtures to account for the conditional rendering- if there aren't any applications then a p tag will tell the user what's going on. If there are (ie the petApplicationsPage fixture) then the applications will be displayed.

File: cypress/integration/pet_applications_spec.js [hunk 10]
```diff
@@ -1,20 +1,61 @@
 describe('Pet Applications Component', () => {
+  beforeEach(() => {
```
Comment (anewb87) [inline cypress/integration/pet_applications_spec.js:2]: Super cool doing the double intercept.

**Review by anewb87 — APPROVED** [review]
Nice call remembering to test before the commit.

**Pull request merged**